tools/lsp/ui/component-list.slint

```diff
@@ -46,25 +54,39 @@ export component ComponentList {
                         }
                     }
 
-                    for ci in cli.components: TouchArea {
+                    for ci in cli.components: component := TouchArea {
                         private property <length> drop-x: self.absolute-position.x + self.mouse-x - root.preview-area-position-x;
                         private property <length> drop-y: self.absolute-position.y + self.mouse-y - root.preview-area-position-y;
                         private property <bool> on-drop-area:
                             drop-x >= 0 && drop-x <= root.preview-area-width && drop-y >= 0 && drop-y <= root.preview-area-height;
-                        private property <bool> can-drop-here: root.can-drop(ci, drop-x, drop-y, on-drop-area);
+                        private property <bool> can-drop-here: !self.data.is-currently-shown && root.can-drop(self.data.name, drop-x, drop-y, on-drop-area);
+                        private property <ComponentItem> data: ci;
+                        private property <bool> edit-component-name;
 
                         enabled: root.preview-visible;
-                        height: name.preferred-height + 10px;
```

Comment:
By removing this binding you are breaking the size constraint.

You can either use a layout and put the LineEdit and Text into it, but that will change then the condition change.
Or you keep the text always instentiated and just make it `visible: !(self.data.is-user-defined && self.edit-component-name)`